core/src/main/java/io/questdb/cairo/ColumnType.java

```diff
@@ -240,8 +248,96 @@ public static int getGeoHashTypeWithBits(int bits) {
         return mkGeoHashType(bits, (short) (GEOBYTE + pow2SizeOfBits(bits)));
     }
 
+    public static int getIntervalType(int timestampType) {
+        assert isTimestamp(timestampType);
+        switch (timestampType) {
+            case TIMESTAMP_MICRO:
+                return INTERVAL_TIMESTAMP_MICRO;
+            case TIMESTAMP_NANO:
+                return INTERVAL_TIMESTAMP_NANO;
+            default:
+                return UNDEFINED;
+        }
+    }
+
+    public static TimestampDriver getTimestampDriver(int timestampType) {
+        final short type = tagOf(timestampType);
+        // todo null and UNDEFINED use MicrosTimestamp
+        if (type == ColumnType.NULL || type == UNDEFINED) {
+            return MicrosTimestampDriver.INSTANCE;
+        }
+
+        assert tagOf(timestampType) == TIMESTAMP;
+        switch (timestampType & TIMESTAMP_TYPE_MASK) {
+            case TIMESTAMP_MICRO:
+                return MicrosTimestampDriver.INSTANCE;
+            case TIMESTAMP_NANO:
+                return NanosTimestampDriver.INSTANCE;
+            default:
+                throw new UnsupportedOperationException();
+        }
+    }
+
+    public static TimestampDriver getTimestampDriverByIntervalType(int intervalType) {
+        assert intervalType == INTERVAL_TIMESTAMP_MICRO || intervalType == INTERVAL_TIMESTAMP_NANO;
+        if (intervalType == INTERVAL_TIMESTAMP_MICRO) {
+            return MicrosTimestampDriver.INSTANCE;
+        } else {
+            return NanosTimestampDriver.INSTANCE;
+        }
+    }
+
+    public static int getTimestampType(int left, int right, CairoConfiguration configuration) {
+        left = getTimestampType(left, configuration);
+        right = getTimestampType(right, configuration);
+        return Math.max(left, right);
```

Comment:
this is looks like a hack.

- we should have explicit priority table, independent of order of the int values of the types
- this should leave near "generate time series" code as a helper. It is highly confusing to have this method on ColumnType API.

Reply:
fixed to `getHigherPrecisionTimestampType`